src/pages/signin/SignInPage.js

```diff
@@ -95,42 +110,37 @@ class SignInPage extends Component {
         // - A login has been entered or a validateCode has been cached from sign in link
         // - AND the login isn't an unvalidated secondary login
         // - AND the user is on the 'passwordless' beta
-        const showValidateCodeForm = (this.props.credentials.login || this.props.credentials.validateCode) && !showUnlinkLoginForm && Permissions.canUsePasswordlessLogins(this.props.betas);
+        const showValidateCodeForm =
+            !showWelcomeForm && (this.props.credentials.login || this.props.credentials.validateCode) && !showUnlinkLoginForm && Permissions.canUsePasswordlessLogins(this.props.betas);
 
         // Show the resend validation link form if
         // - A login has been entered
         // - AND is not validated or password is forgotten
         // - AND the login isn't an unvalidated secondary login
         // - AND user is not on 'passwordless' beta
         const showResendValidationForm =
+            !showWelcomeForm &&
             Boolean(this.props.credentials.login) &&
             (!this.props.account.validated || this.props.account.forgotPassword) &&
             !showUnlinkLoginForm &&
             !Permissions.canUsePasswordlessLogins(this.props.betas);
 
         let welcomeHeader = '';
         let welcomeText = '';
+
+        const userLogin = Str.removeSMSDomain(lodashGet(this.props, 'credentials.login', ''));
+        const userLoginToDisplay = Str.isSMSLogin(userLogin) ? this.props.formatPhoneNumber(userLogin).replace(/ /g, '\u00A0') : userLogin;
+
         if (showValidateCodeForm) {
             if (this.props.account.requiresTwoFactorAuth) {
                 // We will only know this after a user signs in successfully, without their 2FA code
                 welcomeHeader = this.props.isSmallScreenWidth ? '' : this.props.translate('welcomeText.welcomeBack');
                 welcomeText = this.props.translate('validateCodeForm.enterAuthenticatorCode');
-            } else {
-                const userLogin = Str.removeSMSDomain(lodashGet(this.props, 'credentials.login', ''));
-
-                // replacing spaces with "hard spaces" to prevent breaking the number
```

Comment:
Can we add this comment back in? Seemed like useful context since it's kind of unclear why we are doing the replace...